src/game/start.py

```diff
@@ -1,46 +1,48 @@
-from .data import Piece, GameTurn
+import sys
+import getopt
+import json
+from .data import Piece, GameTurn, GameState
 
 
 def start_game():
     print("Welcome to Quarto-Py")
 
+    parameter, error_message = get_state_parameter(sys.argv)
     pieces_list = Piece.create_pieces_list()
 
-    grid = init_grid()
-    remaining_pieces = init_remaining_pieces()
-    game_turn = init_game_turn()
+    game_state = GameState(parameter)
+    if len(game_state.message) > 0:
+        error_message = game_state.message
 
-    display_game(grid, remaining_pieces, game_turn)
+    if len(error_message) > 0:
+        print(error_message)
+        print("""Valid sample : --state='{"grid" : {"A2": 10,"C1":3,"D1":12},"turn" :{"player" : 1,"selected" : 7}}'""")
 
-
-def init_grid(grid_size=4):
-    grid = []
-    i = 0
-    while i < grid_size:
-        grid.append([])
-        j = 0
-        while j < grid_size:
-            grid[i].append('.')
-            j += 1
-        i += 1
-    return grid
+    display_game(game_state)
 
 
-def init_remaining_pieces(pieces_number=16):
-    list_pieces = []
-    i = 0
-    while i < pieces_number:
-        i += 1
-        list_pieces.append(i)
-    return list_pieces
+def get_state_parameter(argv):
+    error_message = ""
+    if 'quarto.py' in argv[0]:
+        argv.pop(0)
 
+    try:
+        opts, args = getopt.getopt(argv, "s:", ["state="])
+    except getopt.GetoptError:
+        parameter = ""
+    for opt, arg in opts:
+        if opt == "--state":
+            parameter = arg
+    try:
+        parameter = json.loads(parameter)
+    except:
+        parameter = ""
+        error_message = "[The state to load is not wellformed] : Ignored"
 
-def init_game_turn():
-    game_turn = GameTurn()
-    return game_turn
+    return parameter, error_message
```

Comment:
This is the Go ways to handle errors.
Please use named / custom exceptions to handle this kind of errors